server/services/store/sqlstore/board.go

```diff
@@ -234,36 +234,6 @@ func (s *SQLStore) getBoard(db sq.BaseRunner, boardID string) (*model.Board, err
 	return s.getBoardByCondition(db, sq.Eq{"id": boardID})
 }
 
-func (s *SQLStore) getBoardsForUserAndTeam(db sq.BaseRunner, userID, teamID string, includePublicBoards bool) ([]*model.Board, error) {
```

Comment:
Once this is removed, we can get rid of the `mattermostauthlayer` dir by bring its code over here. Up to you if you wanna handle that in this PR or create a separate PR for it. If handling in a separate PR, can you please create a Jira ticket to track it?

Reply:
I will take care of this in subsequent PR.
Created a ticket for this: https://mattermost.atlassian.net/browse/MM-59349